pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -277,6 +278,12 @@ func (p *staticPolicy) updateCPUsToReuse(pod *v1.Pod, container *v1.Container, c
 	// If so, add its cpuset to the cpuset of reusable CPUs for any new allocations.
 	for _, initContainer := range pod.Spec.InitContainers {
 		if container.Name == initContainer.Name {
+			if types.IsRestartableInitContainer(&initContainer) {
+				// If the container is a restartable init container, we should not
+				// reuse its cpuset, as a restartable init container can run with
+				// regular containers.
+				break
+			}
```

Comment:
I wonder if a `continue` would be more appropriate here. 

Are we confident that the order is such that (normal/non-restartable) init containers always appear before restartable init containers and are stored in `pod.Spec.InitContainers` in that order? The reason I am asking that is because in case there is some discrepancy in the order (either in the way they are specified or stored) we risk running into a scenario where once we see the first restartable init container we break from the loop and the rest of the initcontainers are skipped even if they are non-restartable where `cpusToReuse` should have been updated.

IOW, we seem to be heavily relying on the assumption that once a restartable init container appears we are guaranteed to not see a normal/non-restartable initcontainer. Perhaps it would be safer to move to the next iteration with the help of a `continue` rather than breaking from loop altogether in order to evaluate all the initcontainers independent of their order.

Reply:
Actually, this function is called with the specific container.
We break here not because it is a restartable init container but because we find the target container.

Let me know if I misunderstand your concern.

Reply:
You understood my question perfectly. Thanks for the clarification. I thought that we were iterating through all the initcontainers to reclaim back resources from all the normal/non-restartable initcontainers as opposed reclaiming resources from a specific container.